src/api/app/controllers/build/file_controller.rb

```diff
@@ -0,0 +1,124 @@
+module Build
+  class FileController < ApplicationController
+    before_action :check_permissions_on_project
+    before_action :require_parameters
+
+    # GET /build/:project/:repository/:arch/:package/:filename
+    def show
+      if regexp
+        process_regexp
+      else
+        pass_to_backend path
+      end
+    end
+
+    # PUT /build/:project/:repository/:arch/:package/:filename
+    def update
+      if regexp
+        process_regexp
+      else
+        unless User.current.is_admin?
+          # this route can be used publish binaries without history changes in sources
+          render_error status: 403, errorcode: "upload_binary_no_permission",
+            message: "No permission to upload binaries."
+          return
+        end
+
+        pass_to_backend path
+      end
+    end
+
+    # DELETE /build/:project/:repository/:arch/:package/:filename
+    def destroy
+      unless permissions.project_change? params[:project]
+        render_error status: 403, errorcode: "delete_binary_no_permission",
+          message: "No permission to delete binaries from project #{params[:project]}"
+        return
+      end
+
+      if params[:package] == "_repository"
+        pass_to_backend
+      else
+        render_error status: 400, errorcode: "invalid_operation",
+          message: "Delete operation of build results is not allowed"
+      end
+
+      return
+    end
+
+    private
+
+    def require_parameters
+      required_parameters :project, :repository, :arch, :package, :filename
+    end
+
+    def prj
+      @prj ||=
+        if params[:package] == "_repository"
+          Project.get_by_name params[:project]
+        else
+          pkg = Package.get_by_project_and_name params[:project], params[:package], use_source: false, follow_multibuild: true
+          pkg.project if pkg.class == Package
```

Comment:
Looks like ```get_by_project_and_name``` either returns nil or a package. It might be better to check for nil here.

Are the exceptions that could be raised by ```get_by_project_and_name``` handled somewhere?